hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -882,17 +895,28 @@ private Response createMultipartKey(OzoneVolume volume, String bucket,
               .createMultipartKey(ozoneBucket, key, length, partNumber,
                   uploadID, chunkSize, (DigestInputStream) body);
         }
-        ozoneOutputStream = getClientProtocol().createMultipartKey(
-            volume.getName(), bucket, key, length, partNumber, uploadID);
 
         if (copyHeader != null) {
           Pair<String, String> result = parseSourceHeader(copyHeader);
-
           String sourceBucket = result.getLeft();
           String sourceKey = result.getRight();
 
           OzoneKeyDetails sourceKeyDetails = getClientProtocol().getKeyDetails(
               volume.getName(), sourceBucket, sourceKey);
+          String range =
+              headers.getHeaderString(COPY_SOURCE_HEADER_RANGE);
+          if (range != null) {
+            RangeHeader rangeHeader =
```

Comment:
nit: Can move `rangeHeader` declaration out of the if block to L908 and reuse `rangeHeader` in the block at L933.

Reply:
Done.